pkgs/build-support/cc-wrapper/default.nix

```diff
@@ -594,11 +595,17 @@ stdenvNoCC.mkDerivation {
     stdenvNoCC.mkDerivation {
       name = "win-dll-hook.sh";
       dontUnpack = true;
-      installPhase = ''
-        echo addToSearchPath "LINK_DLL_FOLDERS" "${cc_solib}/lib" > $out
-        echo addToSearchPath "LINK_DLL_FOLDERS" "${cc_solib}/lib64" >> $out
-        echo addToSearchPath "LINK_DLL_FOLDERS" "${cc_solib}/lib32" >> $out
-      '';
+      installPhase =
+        if targetPlatform.isCygwin then
+          ''
+            echo addToSearchPath "LINK_DLL_FOLDERS" "${cc_bin}/lib" >> $out
+          ''
+        else
+          ''
+            echo addToSearchPath "LINK_DLL_FOLDERS" "${cc_solib}/lib" > $out
+            echo addToSearchPath "LINK_DLL_FOLDERS" "${cc_solib}/lib64" >> $out
+            echo addToSearchPath "LINK_DLL_FOLDERS" "${cc_solib}/lib32" >> $out
+          '';
```

Comment:
Curious what motivated this

Reply:
I don't think I'm convinced. My main motivation for packaging cygwin1.dll was to get bash.exe compiling. I don't really care how many cygwin1.dll instances are being used independently.

Reply:
The problem is you wouldn't be able to mix things (e.g. `nix run nixpkgs-unsable#foo`) that use different cygwin dlls without bad things happening.

Reply:
How come? (Actually curious.) Do the instances of the same DLL manage to share state somehow?

Reply:
https://cygwin.com/faq.html#faq.using.multiple-copies

This probably explains it better than I could.  Even if we changed how those things work, I don't think we could support multiple versions sharing state.

That's why I like to think of as the kernel.